### PR TITLE
[XLATensor] Make convolution attributes more general

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: torch_xla/csrc/ops/conv2d.cpp [hunk 1]
```diff
@@ -37,43 +35,46 @@ xla::PrecisionConfig::Precision MakePrecisionConfig(
 }  // namespace
 
 Conv2d::Conv2d(const NodeOperand& input, const NodeOperand& weight,
-               const NodeOperand& bias, int stride, int padding,
+               const NodeOperand& bias,
+               tensorflow::gtl::ArraySlice<xla::int64> stride,
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/conv2d.cpp]: Let's use 'const T' for array slices.

File: test/cpp/test_tensor.cpp [hunk 2]
```diff
@@ -43,5 +43,95 @@ TEST(TensorTest, TestIntegerAdd) {
...
+                                           /*stride=*/{stride, stride},
+                                           /*padding=*/{padding, padding},
+                                           /*use_full_conv_precision=*/true);
+          } else {
+            dev_output = dev_input->conv2d(*dev_weight,
+                                           /*stride=*/{stride, stride},
+                                           /*padding=*/{padding, padding},
+                                           /*use_full_conv_precision=*/true);
+          }
+
```
Comment (dlibenzi) [inline test/cpp/test_tensor.cpp]: line--?

File: torch_xla/csrc/ops/conv2d.cpp (hunk 1 above)
Comment (dlibenzi) [inline torch_xla/csrc/ops/conv2d.cpp]: const xla::int64

File: torch_xla/csrc/tensor.h [hunk 3]
```diff
@@ -135,9 +135,15 @@ class XLATensor {
 
   std::shared_ptr<XLATensor> threshold(float threshold, float value);
 
-  std::shared_ptr<XLATensor> conv2d(const std::shared_ptr<XLATensor>& weight,
-                                    const std::shared_ptr<XLATensor>& bias,
-                                    int stride, int padding,
+  std::shared_ptr<XLATensor> conv2d(const XLATensor& weight,
+                                    const XLATensor& bias,
+                                    const std::vector<xla::int64>& stride,
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor.h]: Can these be array slices <const int64>?

**Comment by asuhan** [comment]
All comments addressed.

**Pull request merged**